src/dialogflow/sessions_client.rs

```diff
@@ -54,8 +54,8 @@ impl SessionsClient {
 
     /// Returns sender than can be used to stream in audio bytes.
     pub fn get_audio_sink(&mut self) -> Option<mpsc::Sender<StreamingDetectIntentRequest>> {
-        if let Some(audio_sender) = &self.audio_sender {
-            Some(audio_sender.clone())
```

Comment:
ok so when I clone the sender, once all data are sent into stream there will be timeout error(**_code: 11, message: "Audio Timeout Error: Long duration elapsed without audio. Audio should be sent close to real time._**)

Reply:
Exactly.
In order not to have this error occur we need to:
1. Drop the only copy of the sender
2. Be using the latest version of tonic

I made a bin that was basically your example, and the error didn't come through even with the receiver still listening 🤷